src/blueprints/tv2/action-factories/tv2-remote-action-factory.ts

```diff
@@ -116,4 +131,108 @@ export class Tv2RemoteActionFactory {
+      },
+      data: {
+        partInterface: {} as PartInterface, // Is determined when called.
+        pieceInterfaces: [] as PieceInterface[], // Is determined when called.
+      }
+    }
+  }
+
+  public isRemoteAction(action: Action): action is Tv2RemoteAction {
+    const tv2Action: Tv2Action = action as Tv2Action
+    return tv2Action.metadata.contentType === Tv2ActionContentType.REMOTE
+  }
+
+  public getMutateActionMethods(action: Tv2Action): MutateActionMethods[] {
+    switch (action.metadata.actionSubtype) {
+      case Tv2ActionSubtype.RECALL_LAST_REMOTE:
+        return this.getRecallLastRemoteMutateActions()
+
+      default:
+        return []
+    }
+  }
+
+  private getRecallLastRemoteMutateActions(): MutateActionMethods[] {
+    return [
+      {
+        type: MutateActionType.HISTORIC_PART,
+        updateActionWithPartData: this.updateInsertLastVideoClipToInputAction.bind(this),
+        partPredicate: (part: Part) => this.doesPartContainARemotePiece(part),
+      }
+    ]
+  }
+
+  private updateInsertLastVideoClipToInputAction(action: Action, historicPart: Part): Tv2RecallLastRemoteAsNextAction {
+    const clonedPart: Part = historicPart.clone()
+    clonedPart.reset()
+
+    const partInterface: PartInterface = {
+      id: `recall_last_remote_part_${clonedPart.id}`,
+      name: clonedPart.name,
+      segmentId: '',
+      rank: -1,
+      isPlanned: false,
+      isOnAir: false,
+      isNext: false,
+      isUnsynced: false,
+      inTransition: {
+        keepPreviousPartAliveDuration: 0,
+        delayPiecesDuration: 0,
+      },
+      outTransition: {
+        keepAliveDuration: 0,
+      },
+      disableNextInTransition: false,
+      pieces: [],
+    }
+
+    const pieceInterfaces: PieceInterface[] = historicPart.getPieces().map(piece => ({
+      id: `recall_last_remote_piece_${piece.id}`,
+      partId: partInterface.id,
+      name: piece.name,
+      layer: piece.layer,
+      pieceLifespan: piece.pieceLifespan,
+      transitionType: piece.transitionType,
+      isPlanned: false,
+      start: piece.getStart(),
+      duration: piece.duration,
+      preRollDuration: piece.preRollDuration,
+      postRollDuration: piece.postRollDuration,
+      metadata: piece.metadata,
+      tags: [],
+      isUnsynced: false,
+      timelineObjects: piece.timelineObjects,
+    }))
+
+    const recallLastRemoteAction: Tv2RecallLastRemoteAsNextAction = action as Tv2RecallLastRemoteAsNextAction
+    return {
+      ...recallLastRemoteAction,
+      data: {
+        partInterface,
+        pieceInterfaces,
+      }
+    }
+  }
+
+  private doesPartContainARemotePiece(part: Part): boolean {
+    return part.getPieces().some(this.isRemotePiece.bind(this))
+  }
+
+  private isRemotePiece(piece: Piece): boolean {
+    const metadata: Tv2PieceMetadata | undefined = piece.metadata as Tv2PieceMetadata | undefined
+    return metadata?.type === Tv2PieceType.REMOTE
+  }
```

Comment:
Do you really need to bind `this`? You are not using `this` in `isRemotePiece`.

Reply:
Lambda is used instead.